### PR TITLE
Add metadata query operations to ChromaDocumentStore

### LINKED ISSUE #2644: add the following operations to `ChromaDocumentStore`
```python
def count_documents_by_filter(filter: dict) -> int
```
- applies a filter and counts the documents that matched it.

---

```python
def count_unique_metadata_by_filter(filters: dict[str, Any], metadata_fields: list[str]) -> dict[str, int]` 
```
- applies a filter selecting documents and counts the unique values for each meta field of the matched documents

---

```python
def get_metadata_fields_info() -> dict[str, dict]
```
- returns the meta fields and the corresponding types, it should return the same format as in the `OpenSearchDocumentStore`

```python
{
    'content': {'type': 'text'},
    'category': {'type': 'keyword'},
    'status': {'type': 'keyword'},
    'priority': {'type': 'long'},
}
```

---

```python
def get_metadata_field_min_max(metadata_field: str) -> dict[str, Any]
```
- for a given metadata field, find its max and min value

---

```python
def get_metadata_field_unique_values(metadata_field: str, search_term: str | None, from_: int, size: int) -> tuple[list[str], int]:
```
- retrieves unique values for a field matching a search_term or all possible values if no search term is given

---

```python
def query_sql(query: str)
```
- (optional) issue an SQL query for the doc stores that support it

### PR DESCRIPTION
**Pull request by saakshigupta2002**
## Summary

Implements 5 new metadata query operations for ChromaDocumentStore to bring feature parity with other Haystack document stores.

Fixes #2644

### New Methods
- ✅ `count_documents_by_filter()` + async version
- ✅ `count_unique_metadata_by_filter()` + async version
- ✅ `get_metadata_fields_info()` + async version
- ✅ `get_metadata_field_min_max()` + async version
- ✅ `get_metadata_field_unique_values()` + async version

### Helper Methods
- `_normalize_metadata_field_name()` - Normalizes field names (removes "meta." prefix)
- `_infer_type_from_value()` - Infers type from value for schema info

## Implementation Approach

Since ChromaDB lacks native aggregation APIs (unlike OpenSearch), these operations are implemented via client-side aggregation:
1. Fetch filtered documents with minimal data (metadata only)
2. Process metadata in Python (sets for unique counts, min/max for ranges)
3. Apply pagination and return results

This approach is documented in docstrings with performance considerations noted.

## Testing

- ✅ Comprehensive unit tests for all 5 methods (sync + async)
- ✅ Test coverage includes: basic functionality, edge cases, filters, pagination, type handling
- ✅ Tests cover empty collections, missing fields, field normalization, and boundary conditions

## Example Usage

```python
from haystack import Document
from haystack_integrations.document_stores.chroma import ChromaDocumentStore

store = ChromaDocumentStore()
store.write_documents([
    Document(content="Doc 1", meta={"category": "A", "priority": 1}),
    Document(content="Doc 2", meta={"category": "B", "priority": 2}),
])

# Count filtered documents
count = store.count_documents_by_filter({"field": "category", "operator": "==", "value": "A"})
# Returns: 1

# Count unique metadata values
unique_counts = store.count_unique_metadata_by_filter({}, ["category"])
# Returns: {"category": 2}

# Get metadata schema
fields_info = store.get_metadata_fields_info()
# Returns: {"category": {"type": "keyword"}, "priority": {"type": "long"}}

# Get min/max values
min_max = store.get_metadata_field_min_max("priority")
# Returns: {"min": 1, "max": 2}

# Get unique values with pagination
values, total = store.get_metadata_field_unique_values("category", None, 0, 10)
# Returns: (["A", "B"], 2)
```

## Notes

- All methods support both sync and async execution
- Field names can be specified with or without "meta." prefix
- None values are excluded from all calculations
- Performance note: For large collections, client-side aggregation may be slower than database-native aggregations

## Checklist

- [x] Implementation follows existing code patterns
- [x] All methods have comprehensive docstrings
- [x] All methods have complete type hints
- [x] Unit tests cover all functionality
- [x] No breaking changes

### CODE REVIEW

File: integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py [hunk 1]
```diff
@@ -864,6 +899,444 @@ async def search_embeddings_async(
...
+        # Count unique values for each field
+        unique_counts = {}
+        for field in normalized_fields:
+            # Collect all values for this field (skip None values)
+            values = {meta.get(field) for meta in metadatas if meta and field in meta and meta.get(field) is not None}
+            unique_counts[field] = len(values)
+
+        return unique_counts
+
+    async def count_unique_metadata_by_filter_async(
```
Comment (davidsbatista) [inline integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py:1085]: Both this one and the sync version share some code that can be refactored into a helper function. It would be great if you could do it.

File: integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py [hunk 2]
```diff
@@ -864,6 +899,444 @@ async def search_embeddings_async(
...
+            if not meta:
+                continue
+
+            for field_name, value in meta.items():
+                if value is not None and field_name not in field_info:
+                    field_info[field_name] = {"type": self._infer_type_from_value(value)}
+
+        return field_info
+
+    async def get_metadata_fields_info_async(self) -> dict[str, dict[str, str]]:
```
Comment (davidsbatista) [inline integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py:1144]: same thing for this one and it's `sync` version - let's avoid the duplicated code

File: integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py [hunk 3]
```diff
@@ -864,6 +899,444 @@ async def search_embeddings_async(
...
+                val = meta.get(field_name)
+                if isinstance(val, (str, int, float)):
+                    values.append(val)
+
+        if not values:
+            return {"min": None, "max": None}
+
+        return {"min": min(values), "max": max(values)}
+
+    async def get_metadata_field_min_max_async(self, metadata_field: str) -> dict[str, Any]:
```
Comment (davidsbatista) [inline integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py:1196]: same thing for these two

File: integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py [hunk 4]
```diff
@@ -864,6 +899,444 @@ async def search_embeddings_async(
...
+        # Get total count
+        total_count = len(sorted_values)
+
+        # Apply pagination
+        end = from_ + size
+        paginated_values = sorted_values[from_:end]
+
+        return (paginated_values, total_count)
+
+    async def get_metadata_field_unique_values_async(
```
Comment (davidsbatista) [inline integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py:1247]: The same thing for those too as well

File: integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py [hunk 5]
```diff
@@ -864,6 +899,444 @@ async def search_embeddings_async(
...
+        sorted_values = sorted(unique_values)
+
+        # Get total count
+        total_count = len(sorted_values)
+
+        # Apply pagination
+        end = from_ + size
+        paginated_values = sorted_values[from_:end]
+
+        return (paginated_values, total_count)
```
Comment (davidsbatista) [inline integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py]: not: you can remove the parenthesis from all the return statements

**Review by davidsbatista — COMMENTED** [review]
Thanks again for one more contribution! I left some initial comments
